routes/app.js

```diff
@@ -37,14 +37,17 @@ module.exports = function () {
 
     router.get('/admin', authHandler.isAuthenticated, function (req, res) {
         res.render('app/admin', {
-            admin: (req.user.role == 'admin')
+            admin: (req.user.role == 'admin'),
+            securityRating: req.query.securityRating
         })
     })
 
-    router.get('/admin/usersapi', authHandler.isAuthenticated, appHandler.listUsersAPI)
+    router.get('/admin/usersapi/', authHandler.isAuthenticated, authHandler.isAdmin, appHandler.listUsersAPI)
```

Comment:
If `isAdmin` simply tells you if the requester is an admin, it would be up to the callback function (`listUsersAPI`) to decide what to do with that info.

Reply:
No.  If the isAdmin tells you that the requester is an admin, it would call the listUsersAPI via next().  However, if the isAdmin tells you the the requester is NOT an admin, it would return a 401 status, and listUsersAPI would never get called.

In steps: 
1) first is what URL to open
2) isAuthenticated checks if the user is authenticated (correct login, pw) - If yes, go to step 3.  If no, escape to login page.
3) isAdmin check if the user is authorized (isAdmin) - If yes, go to step 4.  If no, escape to 401 status.
4) listUsersAPI returns DB info based on security level.

It's an extra level of security to make sure that an endpoint URL never actually calls the function that does API call (i.e., listUserAPI).  It's the crux of A5 fix.  Hope that helps!

Reply:
Just in case, one more note - This is why the separate security rating also exist for isAdmin.  

From what I can gather in A3, security rating at listUsersAPI controls how much information becomes available to the user (e.g., only id, name, email vs.  all fields in DB)

In A5, the security rating in isAdmin (with the subsequent isAuthenticatedAdmin) controls whether or not listUsersAPI is called at all (via using either next or 401)

So A3 and A5 security levels will do significantly different things.